### PR TITLE
feat(metrics): Add cumulative transaction counters to improve observability

### LINKED ISSUE #12211: Open read-only transactions metric does not work properly for read-only transactions
I currently see the following in the grafana dashboard:
Screenshot 2024-10-30 at 2 45 52 PM

When there are definitely many open transactions. This could be one of two things:
1. The metric is not being updated properly
2. The metric increments / decrements too quickly, and as a result stays at zero.

If 1) is true, the metric should just be fixed.
If 2) is true rather than 1), we might benefit from instead incrementing a counter for cumulative opened transactions, and cumulative closed / aborted transactions. This would also let us track how many read-only transactions are opened in a given time period, for example.

Note: this node was using `--engine.experimental`

### PR DESCRIPTION
**Pull request by hoank101**
Closes #12211

### CODE REVIEW

File: crates/storage/db/src/metrics.rs [hunk 1]
```diff
@@ -258,17 +258,23 @@ impl Labels {
 #[derive(Metrics, Clone)]
 #[metrics(scope = "database.transaction")]
 pub(crate) struct TransactionMetrics {
+    /// Total number of opened database transactions (cumulative)
+    opened_total: Counter,
+    /// Total number of closed database transactions (cumulative)
+    closed_total: Counter,
     /// Total number of currently open database transactions
     open_total: Gauge,
```
Comment (mattsse) [inline crates/storage/db/src/metrics.rs:266]: do we then even need this because this just tracks the diff opened_total - closed_total @Rjected

Comment (Rjected) [reply crates/storage/db/src/metrics.rs:266]: yeah we shouldn't need this now, will need to update the dashboard though

Comment (emhane) [reply crates/storage/db/src/metrics.rs:266]: will you open an issue @Rjected and close this pr and its issue if no longer a problem pls

**Review by onbjerg — CHANGES_REQUESTED** [review]
please also remove `open_total`, and replace any usage of it on the grafana dashboards with `opened_total - closed_total`

**Comment by hoank101** [comment]
hi @onbjerg please help me review it

**Review by mattsse — APPROVED** [review]
lgtm, pending dashboard update

**Review by onbjerg — APPROVED** [review]
lgtm pending dashboard update as well. any progress on this ? @hoank101

**Comment by hoank101** [comment]
hi @onbjerg how to update dashboard? can you show me how

**Review by mattsse — COMMENTED** [review]
looks like we need a new entry here

https://github.com/paradigmxyz/reth/blob/f7dc27f0269bd4c4dfa5b69c84f9b20a1c667251/etc/grafana/dashboards/overview.json#L1727-L1727

also not familiar, so ptal @onbjerg

**Review by emhane — COMMENTED** [review]
cool change!

**Comment by onbjerg** [comment]
See https://github.com/paradigmxyz/reth/tree/main/etc#grafana

**Comment by hoank101** [comment]
i just updated the metrics, @onbjerg please help me review, thanks

**Review by Rjected — APPROVED** [review]
lgtm, thanks! will add a followup issue for making the dashboard track `open - closed`

**Pull request merged**